androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/SocketMessage.kt

```diff
@@ -1,42 +1,25 @@
 package com.mbta.tid.mbta_app.android.util
 
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonPrimitive
 import org.phoenixframework.Message
 
 fun decodeMessage(rawMessage: String): Message {
-    // https://github.com/dsrees/JavaPhoenixClient/blob/1.3.1/src/main/kotlin/org/phoenixframework/Defaults.kt#L68
-    val parseValue: (String) -> String? = { value ->
-        when (value) {
-            "null" -> null
-            else -> value.replace("\"", "")
-        }
-    }
+    val message = Json.parseToJsonElement(rawMessage).jsonArray
+    val joinRef = message[0].jsonPrimitive.content
 
-    var message = rawMessage
-    message = message.removeRange(0, 1) // remove '['
+    val ref = message[1].jsonPrimitive.content
+    val topic = message[2].jsonPrimitive.content
+    val event = message[3].jsonPrimitive.content
 
-    val joinRef = message.takeWhile { it != ',' } // take "join ref", "null" or "\"5\""
-    message = message.removeRange(0, joinRef.length) // remove join ref
-    message = message.removeRange(0, 1) // remove ','
-
-    val ref = message.takeWhile { it != ',' } // take ref, "null" or "\"5\""
-    message = message.removeRange(0, ref.length) // remove ref
-    message = message.removeRange(0, 1) // remove ','
-
-    val topic = message.takeWhile { it != ',' } // take topic, "\"topic\""
-    message = message.removeRange(0, topic.length)
-    message = message.removeRange(0, 1) // remove ','
-
-    val event = message.takeWhile { it != ',' } // take event, "\"phx_reply\""
-    message = message.removeRange(0, event.length)
-    message = message.removeRange(0, 1) // remove ','
-
-    val payloadJson = message.removeRange(message.length - 1, message.length) // remove ']'
+    val payloadJson = message[4].toString()
```

Comment:
I'm nervous about the performance impact of turning this back into a JSON string just so that we can parse it again as JSON a few moments later, but I'm not really sure how straightforward it'd be to avoid that, so it shouldn't have to block this PR.

Reply:
Sure, I don't expect it'll be an issue but I'm digging into Android performance again so I'll keep this in mind as a potential problem. Can always revisit.